### PR TITLE
Add prepare.py replacing index.py

### LINKED ISSUE #26: Please change https://doc.sagemath.org/index.html
Could we make this page 

https://doc.sagemath.org/index.html 

serve
 
$SAGE_ROOT/local/share/doc/sage/index.html

as any other page 

https://doc.sagemath.org/*

serves 

$SAGE_ROOT/local/share/doc/sage/*

Before there was no  $SAGE_ROOT/local/share/doc/sage/index.html file. That seems to be the reason why only this page https://doc.sagemath.org/index.html  serves a custom page. Now that  $SAGE_ROOT/local/share/doc/sage/index.html file looks nice, there is no reason to have an exception.

### PR DESCRIPTION
**Pull request by kwankyu**
Fixes #26 

Configure `prepare.cfg` and then run `./prepare.py`. Note that the script copies the html and pdf files from `$SAGE_ROOT`.

### CODE REVIEW

**Comment by haraldschilly** [comment]
I saw you requested my review. I don't have much time and I don't really know what this even supposed to do. Is there a description somewhere, maybe with a screenshot to compare? How does this differ from the current approach? Just by starting reading the script, it points to a `*.cfg` file, which defines `sage_root: /path/to/sage/root`.  The initial comment says, it takes `$SAGE_ROOT`. So, what do I have to do? Merge this, and then modify the cfg file? Do I have to set the `$SAGE_ROOT` env variable and is this taken into account somehow automatically? (I'm absolutely not familiar with `*.cfg` files).

**Comment by kwankyu** [comment]
> ... So, what do I have to do? Merge this, and then modify the cfg file? 

Yes. Replace "/path/to/sage/root" with the absolute path to your sage root directory

> Do I have to set the `$SAGE_ROOT` env variable and is this taken into account somehow automatically?

No.

Replace "index.py" with "prepape.py" in your workflow of deploying the new doc.

**Comment by haraldschilly** [comment]
Ok, let's see how this plays out. fingers crossed.

The only actual detail I'll change on my end is the copying step. Instead of rsync with a real copy, I do `cp -alR $SAGE_ROOT/local/share/doc/sage/pdf .` for just a hardlink copy. That saves disk space. Looks like there is no reason why not to do this.

The other difference I noticed is the index page. It only points to the main PDFs in the "reference" section. What's up right now also points to all the smaller chapter-wise PDFs. I guess the real question is if those smaller chapter PDFs should even exist in the first place. A question for another day …

File: prepare.py [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+
+show_older_versions = config.getboolean('target', 'show_older_versions')
+
+# ---------------------------------------------------------------
+# copy `$SAGE_ROOT/local/share/doc/sage/html` to `html` directory
+# ---------------------------------------------------------------
+
+log.info('### Copy html documentation')
+
+sp.call(f"rsync -rlp --delete {sage_root}/local/share/doc/sage/pdf ./".split(), stdout=sp.PIPE)
```
Comment (haraldschilly) [inline prepare.py:55]: currently I do `rm directory` + `cp hardlink` instead of rsync. I'll change that on my end.

**Comment by kwankyu** [comment]
Thanks!

**Pull request merged**